core/unit_test/TestGraph.hpp

```diff
@@ -1281,6 +1281,79 @@ TEST(TEST_CATEGORY, mixed_then_host_device_nodes) {
   }
 }
 
+// Ensure that in the default implementation, fencing occurs as needed
+// to ensure that dependencies are met when using an aggregate node.
+//
+// The graph is:
+//
+//              root
+//          (exec_default)
+//               *
+//            *      *
+//    node left      node right
+// (exec_default)  (exec_default)
+//            *      *
+//               *
+//            when_all
+//         (exec_default)
+//               *
+//               *
+//           node final
+//          (exec_other)
+//
+// The default implementation need not fence in the upper part of the graph
+// (diamond) because all nodes are on the same execution space instance.
+// However, before executing 'node final', we must ensure that 'node left' and
+// 'node right' have executed, and so the when_all must be waited for and a
+// fence is needed.
+TEST_F(TEST_CATEGORY_FIXTURE(graph), aggregate_is_awaitable) {
+  const TEST_EXECSPACE exec_default{};
+  const auto execution_space_instances =
+      Kokkos::Experimental::partition_space(exec_default, 1);
+  const auto& exec_other = execution_space_instances.at(0);
+
+  using witness_t = Kokkos::View<int, Kokkos::SharedSpace,
+                                 Kokkos::MemoryTraits<Kokkos::Atomic>>;
+  const witness_t witness(Kokkos::view_alloc("witness", exec_default));
+
+  const auto graph =
+      Kokkos::Experimental::create_graph(exec_default, [&](const auto& root) {
+        const auto node_left = root.then(
+            "node left", exec_default, KOKKOS_LAMBDA() { ++witness(); });
+        const auto node_right = root.then(
+            "node right", exec_default, KOKKOS_LAMBDA() { ++witness(); });
+        Kokkos::Experimental::when_all(node_left, node_right)
+            .then(
+                "node final", exec_other, KOKKOS_LAMBDA() { witness() *= 2; });
+      });
+
+  using namespace Kokkos::Test::Tools;
+  listen_tool_events(Config::DisableAll(), Config::EnableFences());
+
+  if constexpr (is_graph_defaulted<std::remove_cvref_t<decltype(graph)>>) {
```

Comment:
Why are you removing references?
I would have done
```diff
@@ -1185,7 +1185,8 @@ struct GraphIsDefaulted<
 #endif

 template <typename T>
-constexpr bool is_graph_defaulted = GraphIsDefaulted<T>::value;
+constexpr bool is_graph_defaulted =
+    GraphIsDefaulted<std::remove_cv_t<T>>::value;

 // A graph with only one node that is a then_host node.
 TEST(TEST_CATEGORY, then_host) {
```
and just use `declype(graph)` here